### DISCUSSION TITLE
Usage Examples and lightweight/transparent EF abstraction

### DISCUSSION THREAD

**Discussion by austinav** [AutoMapper/AutoMapper.Extensions.ExpressionMapping#169] [General]
I assume the README is showing usage examples and none of the extensions are excerpted from the code. Anyone else have some other usage examples, perhaps to get more separation in repository requests on the basis of Models/DTOs from requiring the repository consumer to know about the EF context/entities? 

For context, something I have been trying to dig into is how to build an abstraction that allows me to write IQueryable<> LINQ against Models/DTOs and let some plumbing (like an IQueryable Provider) handle the translation to EF object model. This expression mapping is a key step in the equation to completely abstracting any custom application code from referencing any EF library objects. Of course, you would have to implement a new provider to switch to a different Data Access/ORM technology, but if you feel the need to protect yourself while getting all the benefits of EF Core, it would seem to come with the territory. Thanks.

**Comment by BlaiseD** [comment]
The [ODATA library](https://github.com/AutoMapper/AutoMapper.Extensions.OData) has quite a few tests should be helpful.  Here's the [typical usage](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/b44c99cfa0afdb2a1ab5a7f4d310ea0ae3f42d90/AutoMapper.AspNetCore.OData.EFCore/QueryableExtensions.cs#L151) of the mapping code.
